tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+)
+
+// Gets the RHCOS base image tag from the BRANCH environment variable or
+// defaults to "latest"; e.g., registry.ci.openshift.org/rhcos-devel/rhel-coreos:<tag>
+func getBaseImageTag() string {
+	branch, found := os.LookupEnv("BRANCH")
+	if !found || branch == "" {
+		return "latest"
+	}
+
+	return strings.ReplaceAll(branch, "release-", "")
+}
+
+// Builds a derived OS image and places it into an Imagesteam. Attempts to use
+// a previously-built image, if one is available.
+func maybeBuildDerivedOSImage(t *testing.T, cs *framework.ClientSet) error {
+	imageStreamConfig := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: mcoNamespace,
+		},
+	}
+
+	_, err := cs.ImageStreams(mcoNamespace).Create(context.TODO(), imageStreamConfig, metav1.CreateOptions{})
+	if err != nil && !k8sErrors.IsAlreadyExists(err) {
+		// If we have an error and it is not an already exists error, something is wrong.
+		return err
+	}
+
+	if k8sErrors.IsAlreadyExists(err) {
+		// We've already got an imagestream matching this name
+		imagestream, err := cs.ImageStreams(mcoNamespace).Get(context.TODO(), imageStreamName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		// Lets see if it has an image we can use
+		if len(imagestream.Status.Tags) != 0 {
+			t.Logf("image already built, reusing!")
+			return nil
+		}
+	}
+
+	baseImageBuildArg := fmt.Sprintf(imagePullSpec, getBaseImageTag())
+
+	t.Logf("imagestream %s created", imageStreamName)
+	t.Logf("using %s as the base image", baseImageBuildArg)
+
+	// Create a new build
+	buildConfig := &buildv1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildName,
+		},
+		Spec: buildv1.BuildSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: "Git",
+					Git: &buildv1.GitBuildSource{
+						URI: "https://github.com/coreos/fcos-derivation-example",
+						Ref: "rhcos",
+					},
+				},
+				Strategy: buildv1.BuildStrategy{
+					DockerStrategy: &buildv1.DockerBuildStrategy{
+						BuildArgs: []corev1.EnvVar{
+							{
+								Name:  "RHEL_COREOS_IMAGE",
+								Value: baseImageBuildArg,
+							},
+						},
+					},
+				},
+				Output: buildv1.BuildOutput{
+					To: &v1.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: imageStreamName + ":latest",
+					},
+				},
+			},
+		},
+	}
+
+	// Start our new build
+	_, err = cs.BuildV1Interface.Builds(mcoNamespace).Create(context.TODO(), buildConfig, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("could not create build: %w", err)
+	}
+
+	t.Log("base image derivation build started")
+
+	build, err := cs.BuildV1Interface.Builds(mcoNamespace).Get(context.TODO(), buildConfig.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get build %s: %w", build.Name, err)
+	}
```

Comment:
Does the `Create` call already return the build?

Reply:
Sort of. I don't think it's a complete build object. In other words, if you didn't supply some metadata, what it returns may be missing that metadata. However, I think that may be fine for what I'm doing with it.